lib/matplotlib/ticker.py

```diff
@@ -1170,25 +1170,208 @@ class LogitFormatter(Formatter):
+                # displayed, then we choose some (spaced) minor ticks to label.
+                # Only minor ticks are known, we assume it is sufficient to
+                # choice which ticks are displayed.
+                # For each ticks we compute the distance between the ticks and
+                # the previous, and between the ticks and the next one. Ticks
+                # with smallest minimum are chosen. As tiebreak, the ticks
+                # with smallest sum is chosen.
+                diff = np.diff(-np.log(1 / self.locs - 1))
+                space_pessimistic = np.minimum(
+                    np.concatenate(((np.inf,), diff)),
+                    np.concatenate((diff, (np.inf,))),
+                )
+                space_sum = (
+                    np.concatenate(((0,), diff))
+                    + np.concatenate((diff, (0,)))
+                )
+                good_minor = sorted(
+                    range(len(self.locs)),
+                    key=lambda i: (space_pessimistic[i], space_sum[i]),
+                )[-self._minor_number:]
+                self._labelled.update(locs[i] for i in good_minor)
+
+    def _format_small_value(self, x, locs):
+        exponent = math.floor(np.log10(x))
+        value = x * 10 ** (-exponent)
+        if len(locs) < 2:
+            precision = 0
+        else:
+            diff = np.sort(np.abs(locs - x))[1]
+            precision = -np.log10(diff) + exponent
+            precision = (
+                int(np.round(precision))
+                if is_close_to_int(precision)
+                else math.ceil(precision)
+            )
+        if precision < 1:
+            precision = 0
+        s = r"%.*f\cdot10^{%d}" % (precision, value, exponent)
         return s
 
+    def _one_minus(self, s):
+        if self._use_overline:
+            return r"\overline{%s}" % s
+        else:
+            return "1-{}".format(s)
+
+    def __call__(self, x, pos=None):
+        if self._minor and x not in self._labelled:
+            return ""
+        if x <= 0 or x >= 1:
+            return ""
+        usetex = rcParams["text.usetex"]
+
+        if is_close_to_int(2 * x) and round(2 * x) == 1:
+            s = self._one_half if self._one_half is not None else "0.5"
+        elif x < 0.5 and is_decade(x, rtol=1e-7):
+            exponent = round(np.log10(x))
+            s = "10^{%d}" % exponent
+        elif x > 0.5 and is_decade(1 - x, rtol=1e-7):
+            exponent = round(np.log10(1 - x))
+            s = self._one_minus("10^{%d}" % exponent)
+        elif x < 0.1:
+            s = self._format_small_value(x, self.locs)
+        elif x > 0.9:
+            s = self._one_minus(
+                    self._format_small_value(1-x, 1-self.locs)
+                )
+        else:
+            if len(self.locs) < 2:
+                precision = 1
+            else:
+                # we look for the nearest ticks from loc (except the tick at
```

Comment:
looks like some of this code is duplicated in _format_small_value?  can it be merged with it?  (dunno)

Reply:
I changed `_format_small_value` to `_format_value` and deduplicate this section. Thanks for notice. Fixed.